### PR TITLE
fix nan handling with zero heatmaps

### PR DESCRIPTION
**Pull request by colehurwitz**
Replace uniform heatmaps with zero heatmaps when labels have NaNs

### CODE REVIEW

**Comment by colehurwitz** [comment]
It passed pytests on my side so I think it should work

File: lightning_pose/data/utils.py [hunk 1]
```diff
@@ -380,6 +379,7 @@ def generate_heatmaps(
     keypoints[:, :, 1] *= out_height / height
     keypoints[:, :, 0] *= out_width / width
     nan_idxs = torch.isnan(keypoints)[:, :, 0]
+    nan_idxs = nan_idxs
```
Comment (danbider) [inline lightning_pose/data/utils.py:382]: not sure about this line, why is it there? should it be deleted?

File: lightning_pose/data/utils.py [hunk 2]
```diff
@@ -358,7 +358,6 @@ def generate_heatmaps(
     width: int,  # width of full sized image
     output_shape: Tuple[int, int],  # dimensions of downsampled heatmap
     sigma: Union[float, int] = 1.25,  # sigma used for generating heatmaps
-    normalize: bool = True,
```
Comment (danbider) [inline lightning_pose/data/utils.py:361]: did we agree to remove the normalize argument @themattinthehatt @colehurwitz ?

File: lightning_pose/data/utils.py [hunk 3]
```diff
@@ -395,18 +395,11 @@ def generate_heatmaps(
...
-
-    if normalize:
-        # normalize all heatmaps to one
-        confidence = confidence / torch.sum(confidence, dim=(2, 3), keepdim=True)
-    else:
-        confidence /= sigma * torch.sqrt(2 * torch.tensor(np.pi))
-
+    # normalize all heatmaps to one
+    confidence = confidence / torch.sum(confidence, dim=(2, 3), keepdim=True)
+    # replace nans with zeros heatmaps
```
Comment (danbider) [inline lightning_pose/data/utils.py:400]: add a comment that "we won't try to output an all-zeros heatmap, but rather all-zeros heatmaps will be ignored in the supervised heatmap loss"

File: lightning_pose/data/utils.py [hunk 4]
```diff
@@ -395,18 +395,11 @@ def generate_heatmaps(
     confidence *= -1
     confidence /= 2 * sigma**2
     confidence = torch.exp(confidence)
```
Comment (danbider) [inline lightning_pose/data/utils.py:397]: @colehurwitz what is `confidence` at this stage for a NaN keypoint? just making sure we have no surprises

**Review by danbider — CHANGES_REQUESTED** [review]
minor suggestions and qs

File: lightning_pose/data/utils.py (hunk 2 above)
Comment (themattinthehatt) [reply lightning_pose/data/utils.py:361]: we discussed this after you left the meeting last week - I don't see any reason why we should have this option available. it isn't used anywhere else. if we need to resurrect it at some point that's fine but for now it's just clutter.

File: lightning_pose/data/utils.py (hunk 4 above)
Comment (themattinthehatt) [reply lightning_pose/data/utils.py:397]: i was thinking about this last time I looked at this function - we should just rename this variable `heatmap` since that's what it actually is

Comment (colehurwitz) [reply lightning_pose/data/utils.py]: Changed the name to heatmap. Honestly, I have no idea what confidence/heatmaps are for NaN points before zeroing.

**Comment by themattinthehatt** [comment]
looks good to me!

**Review by themattinthehatt — APPROVED** [review]

**Review by danbider — APPROVED** [review]
great, happy to have this in
